dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/ParameterUtils.java

```diff
@@ -149,6 +149,33 @@ public static void setInParameter(int index, PreparedStatement stmt, DataType da
         }
     }
 
+    /**
+     * curing user define parameters
+     *
+     * @param globalParamMap  global param map
+     * @param globalParamList global param list
+     * @param commandType     command type
+     * @param scheduleTime    schedule time
+     * @param startParams     start param map
+     * @return curing user define parameters
+     */
+    public static String curingGlobalParams(Map<String, String> globalParamMap, List<Property> globalParamList,
+                                            CommandType commandType, Date scheduleTime, Map<String, String> startParams) {
+        String globalParamsJson = curingGlobalParams(globalParamMap, globalParamList, commandType, scheduleTime);
+        // merge start params
+        if (startParams != null && startParams.size() > 0) {
```

Comment:
I think there may be a problem here, if the value of the parameter set at startup
Is the reference type, eg: (k->time,v->$[yyyy-mm-dd])

Reply:
Thanks for your comment, I don't understand the reference types, would you please show me a JSON sample (maybe global params) which contains this type.

Reply:
If the parameters you set at startup are like this (key->time,value ->$[yyyy-mm-dd]) , it may eventually lead to unresolved

Reply:
You mean the date value? Like {”time”:”2020-01-04”}, I think it should be treated as string value. If I have any misunderstood, please correct me. Thanks.

Reply:
Yes, but in fact, we do use this in some cases, so we need to consider such a situation ((key->time,value ->$[yyyy-mm-dd]),The user used such an expression instead of a fixed value)

Reply:
I think the startup params should be fixed values. If the UI exists, the popup will let user input fixed value, and the fixed value pass into global param. User can set variables in global params, but why does he set the variables in start params?

Reply:
The popup will show when user click start one process.

Reply:
I got it, but you have to consider this situation when you start setting parameters.

Reply:
this is indeed a usage, we always want to be more flexible.

Reply:
I've got your point, user may pass {'biz_date','$[add_months(yyyyMMdd,12*N)]'} to process instance, so the $[add_months(yyyyMMdd,12*N)] should be replaced before merging into intance global params. I will figure out the solution.  I did not know the feature of 'time custom parameter' yesterday, sorry for that.

Reply:
@CalvinKirs I have changed some to support date/time expression. Please have a check.